### ISSUE TITLE
add quantile loss to Gradient Boosted Trees Regressor

### ISSUE THREAD

**Issue by mckeown12** [haifengl/smile#358]
Sklearn's [GBT regressor](http://scikit-learn.org/stable/modules/generated/sklearn.ensemble.GradientBoostingRegressor.html) has the option of choosing a [quantile loss function](https://en.wikipedia.org/wiki/Quantile_regression).  I have found this to be a useful feature which is absent in both smile and sparkml. It sees that the loss functions are all defined in [this file](https://github.com/haifengl/smile/blob/master/core/src/main/java/smile/regression/GradientTreeBoost.java).

**Comment by rayeaster** [comment]
quantile regression objective function with parameter q:
![image](https://user-images.githubusercontent.com/1487283/46842927-06da6580-ce03-11e8-90d4-db8d1cee448b.png)

with q=0.5, it reduced to LAD regression:
![image](https://user-images.githubusercontent.com/1487283/46842957-2f625f80-ce03-11e8-8d23-0e4e8ac6ee83.png)

**Comment by mckeown12** [comment]
I am interested in the $q=.05$ and $q=.95$ cases, giving a sort of confidence interval for prediction.

**Comment by haifengl** [comment]
It is implemented in v2. Thanks
